finance/proposal-calls/cycle3/aperio_fits_dask.md

```diff
@@ -0,0 +1,102 @@
+### Title
+Improve FITS compressed image support and Dask integration in `io.fits`.
+
+### Project Team
+Stuart Mumford
+Thomas Robitaille
+Drew Leonard
+
+### Project Description
+
+This project aims to increase performance of `io.fits` especially around large
+compressed image data.
+
+#### Compressed Image Performance
+
+Currently when using tiled FITS compression the Astropy implementation loads
+all the tiles into RAM and decompresses them all.
+This is highly CPU and memory inefficient, and is one of the reasons Astropy is
+significantly slower at loading these types of files than the `cfitsio` package
+(which uses the same C library as Astropy for this currently).
+
+For example we can compare the loading times of `io.fits` vs `cfitsio`:
+
+Loading a whole array with astropy:
+
+```
+In [44]: %timeit fits.getdata("VBI_L1_00656282_2018_05_11T14_25_05_466665_I.fits", hdu=1)
+183 ms ± 248 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)
+```
+
+with cfitsio:
+```
+In [45]: %timeit hdu[1][:, :]
+131 ms ± 182 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
+```
+
+loading one or more individual tiles with `cfitsio`:
+```
+In [48]: %timeit hdu[1][0, 0]
+21.6 µs ± 191 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
+
+In [49]: %timeit hdu[1][0, :]
+22.3 µs ± 78.2 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
+
+In [50]: %timeit hdu[1][:10, :]
+235 µs ± 718 ns per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
+```
+
+note that doing the same operations with astropy currently loads the whole file
+so takes the same amount of time as the first `getdata` call. This means that
+loading a single tile with astropy is approximately 8300x slower than with
+`cfitsio`!!
+
+Issue [#3895](https://github.com/astropy/astropy/issues/3895) has been open
```

Comment:
This paragraph is good but I think also missing that it should say explicitly that aside from writing the compression in Python one of the big things is obviously to use dask to do on-demand decompression of chunks (as it is right now, this sub-section is missing a 'and we will therefore do this with dask' - I think it would also be good to somewhere mention in a couple of sentences what is dask and why it can help.